src/covariance_parameters/selfadjoint.jl

```diff
@@ -0,0 +1,129 @@
+const RSym{T,S} = Symmetric{T,S} where {T<:Real,S}
+const CHerm{T,S} = Hermitian{T,S} where {T<:Complex,S}
+const SelfAdjoint{T,S} = Union{RSym{T,S},CHerm{T,S}} where {T,S}
```

Comment:
Are RSym and CHerm ever use anywhere? Depending on this it could be simpler to just spell it out:
```suggestion
const RealSymmetric{T,S} = Symmetric{T,S} where {T<:Real,S}
const ComplexHermitian{T,S} = Hermitian{T,S} where {T<:Complex,S}
const SelfAdjoint{T,S} = Union{RealSymmmetric{T,S},ComplexHermitian{T,S}} where {T,S}
```